common/schema/definition.go

```diff
@@ -509,6 +568,16 @@ END`,
 	}.finalize()
 }
 
+func (column *Column) isSelfGenerated() bool {
+	return strings.Contains(column.ClickHouseGenerateFrom, column.Name)
+}
```

Comment:
It should be a specific attribute `ClickHouseSelfGenerated` that should be set on the few columns in this case. While unlikely to have a name of an unrelated column, this is safer.

Reply:
done